gulpfile.js

```diff
@@ -0,0 +1,109 @@
+const { src, dest } = require('gulp');
+const svgmin = require('gulp-svgmin');
+const rename = require('gulp-rename');
+const { obj } = require('through2');
+const ltx = require('ltx');
+const yaml = require('js-yaml');
+const { readFileSync } = require('fs');
+
+// Load SVGO preferences from config file to keep things DRY
+const svgoConfig = yaml.safeLoad(readFileSync('.svgo.yml', 'utf8'));
+
+// Properties to make configurable via Twig templates
+const dynamicSvgProps = [
+  'aria-hidden',
+  'aria-labelledby',
+  'class',
+  'focusable',
+  'height',
+  'id',
+  'preserveAspectRatio',
+  'role',
+  'style',
+  'viewBox',
+  'width'
+];
+
+/**
+ * Accepts SVG source markup and templatizes root attributes while also adding
+ * layout blocks (`before`, `content` and `after`) to its contents.
+ *
+ * @param {String} src - The source SVG markup.
+ * @returns {String}
+ */
+function templatizeSvgString(src) {
+  const svg = ltx.parse(src);
+
+  // Create blocks for SVG content, before and after
+  const prepend = ltx.parse(
+    '<root>{% block before %}{% endblock %}{% block content %}</root>'
+  );
+  const append = ltx.parse(
+    '<root>{% endblock %}{% block after %}{% endblock %}</root>'
+  );
+  svg.children = [...prepend.children, ...svg.children, ...append.children];
+
+  // Identify props already in use in the SVG versus those yet to be used
+  const usedProps = dynamicSvgProps.filter(prop => Boolean(svg.attrs[prop]));
+  const unusedProps = dynamicSvgProps.filter(prop => !svg.attrs[prop]);
+
+  // Properties already in use should have their value set to a conditional.
+  // The `default` filter would be less code, but things get tricky when it
+  // comes to managing quotation marks in XML.
+  usedProps.forEach(prop => {
+    const current = svg.attrs[prop];
+    const twigProp = prop.replace(/-/g, '_');
+    svg.attrs[
+      prop
+    ] = `{% if ${twigProp} %}{{${twigProp}}}{% else %}${current}{% endif %}`;
+  });
```

Comment:
This may be overkill, but I wonder if some props like class should be set up to append content that's passed in instead of replacing old prop content.

Reply:
(I'm fine punting on this)

Reply:
@Paul-Hebert We had this ability in another project, but the more I thought about it the more I wondered if it was really necessary. Eliminating this feature greatly reduced the code this takes, so my instinct was to proceed without it for now.

Reply:
Cool, that works for me :+1: